### PR TITLE
Template for TLS v1.2 only

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Template for TLS v1.2 only

# Testing

```
./configure --enable-usersettings --disable-examples
make
./wolfcrypt/test/testwolfcrypt
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: examples/configs/user_settings_tls12.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+/* Example for TLS v1.2 client only, ECC only, AES GCM only, SHA2-256 only */
+/* Derived using:
+ * ./configure --disable-rsa --disable-dh --disable-tls13 --disable-chacha --disable-poly1305 --disable-sha224 --disable-sha --disable-md5
+ * And generated wolfssl/options.h
+ */
```
Comment (douzzer) [inline examples/configs/user_settings_tls12.h:26]: should add the build+test instructions here.  from the PR comments:
```
./configure --enable-usersettings --disable-examples
make
./wolfcrypt/test/testwolfcrypt
```

File: examples/configs/user_settings_tls12.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#define WOLFSSL_USER_SETTINGS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* ------------------------------------------------------------------------- */
+/* Platform */
+/* ------------------------------------------------------------------------- */
+#define WOLFSSL_USER_IO /* Use the SetIO callbacks, not the internal wolfio.c socket code */
```
Comment (douzzer) [inline examples/configs/user_settings_tls12.h:38]: line length

**Review by douzzer — CHANGES_REQUESTED** [review]
works for me -- just suggested a couple non-functional tweaks.

**Comment by dgarske** [comment]
Retest this please

**Review by douzzer — APPROVED** [review]

**Pull request merged**